app/views/shared/_select_work_type_modal.html.erb

```diff
@@ -11,34 +11,38 @@
           <div class="row match-my-cols">
             <div class="col-md-12">
               <% model_list_fixed.each_slice(2) do |row_presenter| %>
-                  <div class="row">
-                    <div class="col-md-6 select-worktype">
+                <div class="row">
+                  <div class="col-md-6 select-worktype">
+                    <% if row_presenter[0]["visibility"] %>
                       <label>
-                        <input type="radio" name="payload_concern" value="<%= row_presenter[0].concern %>"
-                               data-single="<%= new_polymorphic_path([main_app, row_presenter[0].concern]) %>"
-                               data-batch="<%= hyrax.new_batch_upload_path(payload_concern: row_presenter[0].concern) %>">
+                        <input type="radio" name="payload_concern" value="<%= row_presenter[0]["model"].concern %>"
+                               data-single="<%= new_polymorphic_path([main_app, row_presenter[0]["model"].concern]) %>"
+                               data-batch="<%= hyrax.new_batch_upload_path(payload_concern: row_presenter[0]["model"].concern) %>">
                         <div class="select-work-icon">
-                          <span class="<%= row_presenter[0].icon_class %>"></span>
+                          <span class="<%= row_presenter[0]["model"].icon_class unless row_presenter[0].nil? %>"></span>
```

Comment:
I'd combine the nil? check alongside the visibility check on line #16 and get rid of these hanging `unless row_presenter[0].nil?`.

Also.. it might be worthwhile making a _select_work_type_cell.html.erb partial with lines 15 through 30, passing in each `row_presenter`.